lnbits/core/migrations.py

```diff
@@ -188,3 +192,64 @@ async def m005_balance_check_balance_notify(db):
         );
     """
     )
+
+
+async def m006_add_invoice_expiry_to_apipayments(db):
+    """
+    Adds invoice expiry field to apipayments and precomputes them for
+    existing entries
+    """
+    try:
+        await db.execute("ALTER TABLE apipayments ADD COLUMN expiry TIMESTAMP")
+    except OperationalError:
+        pass
+    try:
+        rows = await (
+            await db.execute(
+                f"""
+                SELECT bolt11, checking_id
+                FROM apipayments
+                WHERE pending = true
+                AND bolt11 IS NOT NULL
+                AND expiry IS NULL
+                AND amount > 0 AND time < {db.timestamp_now}
+                """
+            )
+        ).fetchall()
+        # then we delete all expired invoices, checking one by one
+        print(f"Checking expiry of {len(rows)} invoices")
+        for i, (
+            payment_request,
+            payment_hash,
+        ) in enumerate(rows):
+            print(f"Checking invoice {i}/{len(rows)}")
+            try:
+                invoice = bolt11.decode(payment_request)
+            except:
+                continue
+            if payment_hash != invoice.payment_hash:
+                print("Error: {payment_hash} != {invoice.payment_hash}")
+                continue
+
```

Comment:
- the `continue` statement, means some invoices will have None for `expiry`
- this means the `Payment` model must make `expiry` optional (`expiry: Optional[int]`)